domain/credential/state/state.go

```diff
@@ -220,20 +236,54 @@ ON CONFLICT(cloud_credential_uuid, key) DO UPDATE SET key=excluded.key,
 	return nil
 }
 
+// dbCredentialFromCredential is responsible for populating a database
+// representation of a cloud credential from a credential id and info structures.
+//
+// If no user is found for the credential owner then an error satisfying
+// [usererrors.NotFound] will be returned.
 func dbCredentialFromCredential(ctx context.Context, tx *sqlair.TX, credentialUUID string, id credential.ID, credential credential.CloudCredentialInfo) (*Credential, error) {
 	cred := &Credential{
 		ID:         credentialUUID,
 		Name:       id.Name,
 		AuthTypeID: -1,
 		// TODO(wallyworld) - implement owner as a FK when users are modelled.
-		OwnerUUID:     id.Owner,
+		//OwnerUUID:     id.Owner,
 		Revoked:       credential.Revoked,
 		Invalid:       credential.Invalid,
 		InvalidReason: credential.InvalidReason,
 	}
 
+	userQ := `
+SELECT user.uuid AS &M.userUUID
+FROM user
+WHERE user.name = $M.userName
+AND removed = false
+`
+
+	stmt, err := sqlair.Prepare(userQ, sqlair.M{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var userResult = sqlair.M{}
+	err = tx.Query(ctx, stmt, sqlair.M{"userName": id.Owner}).Get(&userResult)
+	if errors.Is(err, sql.ErrNoRows) {
+		return nil, fmt.Errorf(
+			"%w for name %q as owner for credential %q",
+			usererrors.NotFound,
+			id.Owner,
+			id.Name,
+		)
+	} else if err != nil {
+		return nil, fmt.Errorf(
+			"getting credential %q owner %q: %w",
+			id.Name, id.Owner, err,
+		)
+	}
+	cred.OwnerUUID = userResult["userUUID"].(string)
```

Comment:
Annoyingly this could panic, we should check if this result exists before attempting to convert it to a string if it is nil.